projects/hslayers/src/components/layermanager/layermanager-metadata.service.ts

```diff
@@ -233,13 +251,18 @@ export class HsLayerManagerMetadataService {
       )
         .then((capabilities_xml) => {
           const parser = new WMSCapabilities();
-          const caps = parser.read(capabilities_xml);
-          const paramLayers: string = layer.getSource().getParams().LAYERS;
+          const caps: WMSGetCapabilitiesResponse = parser.read(
+            capabilities_xml
+          );
+          const src = layer.getSource();
+          const params = src.getParams();
+          const layerNameInParams = params.LAYERS;
+          console.log('caps received', caps, layerNameInParams);
 
-          this.parseLayerInfo(layer, paramLayers, caps);
+          this.parseLayerInfo(layer, layerNameInParams, caps, false);
           if (getSubLayers(layer)) {
-            const src = layer.getSource();
-            const params = src.getParams();
+            this.parseLayerInfo(layer, getSubLayers(layer), caps, true);
```

Comment:
We dont need the parsing of LayerInfo twice

Reply:
Thanks for spotting it!

Reply:
Yes, it seems so. Thanks!